File: lib/bodies/content.dart
import 'package:mango_ui/keys.dart';

import 'banner.dart';
import 'colour.dart';
import 'contact.dart';
import 'information.dart';
import 'section.dart';

class Content {
  final String realm;
  final String client;
  final Key logo;
  final String language;
  final String email;
  final Banner banner;
  final Section sectionA;
  final Section sectionB;
  final Information info;
  final Colour colour;
  final List<Contact> contacts;

  Content(
      this.realm,
      this.client,
      this.logo,
      this.language,
      this.email,
      this.banner,
      this.sectionA,
      this.sectionB,
      this.info,
      this.colour,
      this.contacts);

  Map<String, dynamic> toJson() {
    return {
      "Realm": realm,
      "Client": client,
      "LogoKey": logo,
      "Language": language,
      "Banner": banner,
      "SectionA": sectionA,
      "SectionB": sectionB,
      "Info": info,
      "Colour": colour,
      "Email": email,
      "Contacts": contacts
    };
  }
}

File: core/content.go
package core

import (
	"github.com/louisevanderlith/husk/keys"
	"github.com/louisevanderlith/husk/validation"
)

type Content struct {
	Realm       string
	Client      string
	LogoKey     *keys.TimeKey
	Language    string //en,af, en-US
	Banner      Banner
	SectionA    Section
	SectionB    Section
	Info        Information
	Colour      Colour
	Email       string
	Contacts    []Contact
	Description string
	GTag        string
}

func (o Content) Valid() error {
	return validation.Struct(o)
}
